src/Avalonia.Controls/Primitives/Popup.cs

```diff
@@ -860,22 +860,7 @@ private void SetTemplatedParentAndApplyChildTemplates(IControl? control)
         {
             if (control != null)
             {
-                var templatedParent = TemplatedParent;
-
-                if (control.TemplatedParent == null)
-                {
-                    control.SetValue(TemplatedParentProperty, templatedParent);
-                }
-
-                control.ApplyTemplate();
-
-                if (!(control is IPresenter) && control.TemplatedParent == templatedParent)
```

Comment:
@grokys it seems it's sufficient to use logical tree, instead of checking if child is a presenter or not. I.e. we don't need to care about this in TemplatedControl.ApplyTemplatedParent

Reply:
Yeah again, can't remember why it was done like this. If it doesn't break any tests then might not be needed.

Reply:
Ok, I just found something that this broke ;) Going to write a unit test/fix and open a PR.